internal/gatewayapi/helpers.go

```diff
@@ -409,13 +409,18 @@ func routeName(route RouteContext, ruleIdx, matchIdx int) string {
 	return fmt.Sprintf("%s-%s-rule-%d-match-%d", route.GetNamespace(), route.GetName(), ruleIdx, matchIdx)
 }
 
-func irTLSConfig(tlsSecret *v1.Secret) *ir.TLSListenerConfig {
-	if tlsSecret == nil {
+func irTLSConfigs(tlsSecrets []*v1.Secret) []*ir.TLSListenerConfig {
+	if len(tlsSecrets) == 0 {
 		return nil
 	}
 
-	return &ir.TLSListenerConfig{
-		ServerCertificate: tlsSecret.Data[v1.TLSCertKey],
-		PrivateKey:        tlsSecret.Data[v1.TLSPrivateKeyKey],
+	tlsListenerConfigs := make([]*ir.TLSListenerConfig, len(tlsSecrets))
+	for i, tlsSecret := range tlsSecrets {
+		tlsListenerConfigs[i] = &ir.TLSListenerConfig{
+			SecretName:        tlsSecret.Namespace + "-" + tlsSecret.Name,
```

Comment:
nit: good to add a helper function that can be reused later for the name
https://github.com/envoyproxy/gateway/blob/40830e86066e5729ea5719d1561b2e8cafe2dd39/internal/gatewayapi/helpers.go#L392